src/compiler/crystal/compiler.cr

```diff
@@ -502,11 +502,13 @@ module Crystal
       {units, reused}
     end
 
+    private def wants_stats_or_progress?
```

Comment:
**question:** What's the reason for extracting this to a method?
Did you consider duplicating the local variable in different mehtods? Caching the result in a local var has some performance impact, although it's probably negligible.
It might also be worth considering to move this method to `ProgressTracker`.

Reply:
To avoid duplication, or passing a peripheral variable to inner methods.

It would indeed be better in ProgressTracker :+1:

Reply:
Performance impact is likely irrelevant compared to the actual codegen.

We can collect reused methods after the fact in sequential (then MT) codegen. Only the forked codegen needs to collect as we receive results.

Reply:
In fact there are issues: we don't need to collect reused modules on progress, nor do we want to report progress on stats.

Reply:
Fix extracted as #14763